### DISCUSSION TITLE
I need help

### DISCUSSION THREAD

**Discussion by ravenofc** [dyc3/steamguard-cli#343] [General]
Hi I want to run this trough tor network.
I tried on tails and qubes os with whonix and same error is shown after i enter password for the steam acc:

```
ERROR - Unexpected error when trying to log in. If you report this as a bug, please rerun with -v debug or -v trace and include all output in your issue. OtherFailure(NetworkFailure: Transport failed to make request: error sending request for url (https://api.steampowered.com/IAuthenticationService/GetPasswordRSAPublicKey/v1?input_protobuf_encoded=CghzbmFyMTAwMA%3D%3D): error trying to connect: dns error: failed to lookup address information: Name does not resolve

Caused by:
    0: error sending request for url (https://api.steampowered.com/IAuthenticationService/GetPasswordRSAPublicKey/v1?input_protobuf_encoded=CghzbmFyMTAwMA%3D%3D): error trying to connect: dns error: failed to lookup address information: Name does not resolve
    1: error trying to connect: dns error: failed to lookup address information: Name does not resolve
    2: dns error: failed to lookup address information: Name does not resolve
    3: failed to lookup address information: Name does not resolve)
thread 'main' panicked at src/commands/setup.rs:39:14:
Failed to log in. Account has not been linked.: OtherFailure(NetworkFailure: Transport failed to make request: error sending request for url (https://api.steampowered.com/IAuthenticationService/GetPasswordRSAPublicKey/v1?input_protobuf_encoded=CghzbmFyMTAwMA%3D%3D): error trying to connect: dns error: failed to lookup address information: Name does not resolve

Caused by:
    0: error sending request for url (https://api.steampowered.com/IAuthenticationService/GetPasswordRSAPublicKey/v1?input_protobuf_encoded=CghzbmFyMTAwMA%3D%3D): error trying to connect: dns error: failed to lookup address information: Name does not resolve
    1: error trying to connect: dns error: failed to lookup address information: Name does not resolve
    2: dns error: failed to lookup address information: Name does not resolve
    3: failed to lookup address information: Name does not resolve)
note: run with RUST_BACKTRACE=1 environment variable to display a backtrace
```

Obvious this is cause of my network going trough tor, but i cant find a way to make it work and it is not a option for me to use a normal proxy or something, I indeed need to make it work like this. 
This is not a problem on steamguard cli but on my end of course but i really need this to work so i am here looking for help.
Any idea of how to do it?
Thank you

**Comment by dyc3** [comment]
Does `--http-proxy` not work with tor? IIRC, tor exposes a socks proxy which should be compatible

**Reply by ravenofc** [reply]
I did steamguard --http-proxy 127.0.0.1:9050 setup and still error.
Does it support socks5?

**Reply by dyc3** [reply]
Try `steamguard --http-proxy socks5://127.0.0.1:9050`

**Reply by ravenofc** [reply]
I tried that before it gave me unknow proxy scheme error.
I tried `torsocks steamguard setup` and all went trough with no error but i received the email for confirmation and still has my location so it is not working.

**Reply by ravenofc** [reply]
I found a way.
Installed this [Github Link](https://github.com/oyyd/http-proxy-to-socks)
In whonix qube which is already running trough tor
Runned `hpts -s 127.0.0.1:9050 -p 8080`
And used `steamguard --http-proxy 127.0.0.1:8080 setup` 
Setup went trough no errors and email confirmation has random location so the request is going trough tor
